### DISCUSSION TITLE
INFO: Nuevos traductores (Ghrull)

### DISCUSSION THREAD

**Discussion by Zakees** [Ludeon/RimWorld-Spanish#66] [Q&A]
(Tema original de @Ghrull)

Saludos, hago una pequeña guía/sumario de la traducción de Rimword al castellano para ayudar a los nuevos traductores como @colo13verdadeo y que no esten tan perdidos como lo estuve yo en su día.

### Herramientas para traducir:

**Rimtrans:** Esta herramienta genera los archivos *.xml en ingles, a partir de esos archivos hacemos la traducción.
_Enlace: https://github.com/RimWorld-zh/RimTrans_

**Notepad ++:** Con esta herramienta se edita y traduce los archivos *.xml. Hay otros editores y formas de editar, pero personalmente creo que este editor es el mejor y es gratis.
_Enlace: https://notepad-plus-plus.org/_

**Deepl:** El mejor traductor online que hay actualmente, es perfecto para consultar palabras o frases de difícil traducción.
_Enlace: https://www.deepl.com/translator_

**Correctores de ortografía y gramática:** Estas herramientas online pueden ayudar a corregir los textos ya traducidos antes de ser subidos a Github.
_Enlace: https://languagetool.org/es/_
_Enlace: https://spanishchecker.com/_

### Estado de la traducción (13/07/2018):

**Carpeta Backstories:** Solo contiene un archivo (Backstories.xml) que hay que revisarlo entero ya que muchas traducciones están mal, esta desactualizado y hay que añadir la dualidad de genero. Creo que lo mejor seria traducir Backstories.xml directamente del archivo original en ingles. Sobre la dualidad de genero, este seria un ejemplo: 

Actualmente: [PAWN_nameDef] (Por ejemplo, Laura) trabajaba como granjero y era muy feo.
Sustituir por: [PAWN_nameDef] (Por ejemplo, Laura) trabajaba como granjero(a) y era muy feo(a).

Siempre usaremos los paréntesis en lugar de barras puesto que ya se hizo así en los otros textos del juego. 

**Carpeta DefInjected:** Esta carpeta es la que mas textos contiene del juego, esta traducida prácticamente entera pero hay que revisar muchas cosas:

-Las carpetas que a su derecha tienen el texto "Traducción 1.0.1936 rev833 (development build)" o "Traducción 1.0.1938 (development build)" tienen que ser revisadas ya que podrían haber añadido alguna nueva linea o cambio del código, por ejemplo, en ThingDef/Plants_Cultivated_Farm.xml hay que añadir el cacao.

-Las carpetas que a su derecha tienen el texto "Traducción 0.18.1722" tienen que ser revisadas y actualizadas.

-Las carpetas de reglas, por ejemplo RulePackDef, las ha traducido y corregido el compañero de la traducción latina @mora145 , a no ser que sepáis lo que hacéis, recomiendo no modificarlas.

-El resto de carpetas, sobre todo las que no se actualizan desde hace mucho tiempo tienen que ser revisadas y actualizadas (código y texto si fuese necesario).

**Carpeta Keyed:** Los archivos que a su derecha tienen el texto "Traducción 1.0.1938 (development build)" tienen que ser revisados ya que podrían haber añadido alguna nueva linea o cambio del código, el resto de archivos tienen que ser traducidos y actualizados de nuevo ya que están muy desactualizados.

**Carpeta Strings:** Esta carpeta no esta traducida, yo la actualice a la versión 1.0 en ingles porque la versión que estaba traducida era muy antigua y daba muchos problemas. Son traducciones pequeñas y algunos archivos no necesitan ser traducidos.

**LEER PD AL FINAL**

### Consejos:

-Cuando se vaya a actualizar un archivo que este muy anticuado lo mejor es traducir directamente el archivo actualizado de la versión en ingles y luego subir ese archivo (borrar el antiguo y subir el nuevo).

-Puede ser de ayuda usar como referencia la traducción latina ya que esta muy actualizada pero nunca hacer un corta y pega porque hay palabras que ellos usan y nosotros no.

-Si se usa la traducción latina como referencia hay que fijarse si el texto traducido coincide con el texto en original ingles porque hay algunos textos (sobre todo descripciones de objetos) que en las ultimas versiones han cambiado y en la traducción latina no lo han hecho, es el caso de la carpeta DefInjected/ResearchProjectDef/ que en las ultimas versiones cambiaron las descripciones y en la traducción latina siguen las traducciones de los textos antiguos.

-Cuando se suba o modifique algún archivo, en el apartado "Commit changes" seria de gran ayuda escribir la versión del archivo y si es posible una descripción de lo que se a modificado, por ejemplo "Traducción 1.0.1938", de esta manera estará todo mas organizado y sera mas fácil saber que carpetas y archivos necesitan ser actualizados o revisados.

-**REVISAR BIEN LAS FALTAS DE ORTOGRAFÍA Y NO HACER UN CORTA Y PEGA DEL TRADUCTOR.**

**PD:** La traducción al castellano llevaba 2 años sin actualizarse (desde la Alpha 16, los antiguos traductores están inactivos), actualice la mayoría de textos a la versión 1.0 pero al estar traduciendo yo solo (agradecer a @mora145 la ayuda que me a prestado) no va todo lo rápido que me gustaría. Llevo unas semanas bastante ocupado y no he podido hacer mucho, ademas me "queme" un poco porque traducía textos y al día siguiente metían una actualización en la que cambiaban, añadían o quitaban textos a los archivos que acababa de traducir y la mayoría de las veces no sabia exactamente que era lo que habían añadido o quitado porque son muchísimos archivos y tengo que ir comparándolos uno a uno con la versión inglesa, creo que hay una herramienta que puede ayudar con esto pero tengo que consultarlo con los compañeros de la versión latina. Ademas, la herramienta que uso para generar textos (Rimtrans) es de la alpha 18 y da errores con la versión 1.0 (espero que pronto la actualicen). De todas maneras pienso seguir traduciendo Rimworld en mis ratos libres pero seria de gran ayuda que mas traductores se uniesen. Saludos!!

**Comment by Zakees** [comment]
Añadiré, aún habiendo pasado eones desde esta guía, para evitar conflictos. En el tema de la dualidad, en lugar de colocar el paréntesis con la variación [ lo que antes era granjero(a) ] utilizaremos la función **PAWN_gender.** 

De esta manera, el propio juego reconocerá el género y colocará un string u otro según su género, quedando perfectamente establecido.
**Antes:** [PAWN_nameDef] nació enfermizo(a)
**Ahora:** [PAWN_nameDef] nació enfermiz{PAWN_gender ? o : a} --> Si es masculino (es el primer caso siempre) aparecerá enfermizo. Si es femenino, enfermiza.

El juego siempre coloca el primer resultado como el género número 1, masculino; por lo que el 2º es el femenino. 
Hay una distinción, que es en el caso del colono transgenero, que obviamente debe estar al revés.

**Comment by Xeros08** [comment]
El enlace al Discord no es valido. Probablemente esté caducado.

**Reply by jereef** [reply]
Sigue sin funcionar actualmente no?

**Reply by Xeros08** [reply]
Si

**Reply by Zakees** [reply]
El Discord ese dejó de usarse. Lo retiro del topic, aunque podemos ver de hacer uno nuevo.
